cmd/cosign/cli/verify/verify.go

```diff
@@ -136,29 +135,16 @@ func (c *VerifyCommand) Exec(ctx context.Context, images []string) (err error) {
 		co.ClaimVerifier = cosign.SimpleClaimVerifier
 	}
 
-	if c.TSACertChainPath != "" {
-		_, err := os.Stat(c.TSACertChainPath)
-		if err != nil {
-			return fmt.Errorf("unable to open timestamp certificate chain file: %w", err)
-		}
-		// TODO: Add support for TUF certificates.
-		pemBytes, err := os.ReadFile(filepath.Clean(c.TSACertChainPath))
-		if err != nil {
-			return fmt.Errorf("error reading certification chain path file: %w", err)
-		}
-
-		leaves, intermediates, roots, err := tsa.SplitPEMCertificateChain(pemBytes)
-		if err != nil {
-			return fmt.Errorf("error splitting certificates: %w", err)
-		}
-		if len(leaves) > 1 {
-			return fmt.Errorf("certificate chain must contain at most one TSA certificate")
-		}
-		if len(leaves) == 1 {
-			co.TSACertificate = leaves[0]
+	tsaCertificates, err := cosign.GetTSACerts(ctx, c.TSACertChainPath, cosign.GetTufTargets)
+	if err != nil {
+		if c.TSACertChainPath != "" {
+			return fmt.Errorf("unable to load TSA certificates from specified path '%s': %w", c.TSACertChainPath, err)
 		}
-		co.TSAIntermediateCertificates = intermediates
-		co.TSARootCertificates = roots
+		ui.Warnf(ctx, "no TSA certificate chain path provided, or unable to load TSA certificates: %s", err.Error())
```

Comment:
Thinking more about this, I don't think we should add a warning as the vast majority of users aren't using TSA certs and this will just create noise. I'm also not a fan of silently ignoring errors though.

The issue is that we don't know when to look up TSA certs in TUF. For Rekor, we always do unless you provide a flag, and same with Fulcio, we always do unless you provide a key. For TSA certs, it's only going to be for private deployments.

WDYT about adding a flag to control this behavior? If either `--timestamp-certificate-chain=<path>` or `--use-signed-timestamps` is set, then `cosign.GetTSACerts` is invoked. You can then always fail if `err != nil`.

@cmurphy did you have any suggestions?

Reply:
@haydentherapper should i revert back to the following for this PR ?
```
if err != nil {
	ui.Warnf(ctx, fmt.Sprintf("unable to load or get TSA certificates: %s", err.Error()))
}
```

Reply:
FWIW, this is solved in sigstore-go. We load all trust metadata initially, and try to verify signed material with that trust metadata. For timestamps, we only require one to be verifiable.

Reply:
something like this?
- https://github.com/sigstore/cosign/pull/3600/commits/e02e23ffaa7ee19f30aed5fc79ce7d649ee4a410

Reply:
Looks good! Can you do the same for the other verify_*?

Reply:
okay, "i think" i did that right for the other verify files, but it's late in the day here so would welcome the extra 👀:
- https://github.com/sigstore/cosign/pull/3600/commits/ff6d72b8fe98273b2bf7b4c2c6f987e22a0257fd